javaparser-core/src/main/java/com/github/javaparser/ast/type/ClassOrInterfaceType.java

```diff
@@ -53,8 +55,14 @@ public ClassOrInterfaceType(final ClassOrInterfaceType scope, final String name)
     }
 
     public ClassOrInterfaceType(final int beginLine, final int beginColumn, final int endLine, final int endColumn,
-            final ClassOrInterfaceType scope, final String name, final List<Type> typeArgs) {
+                                final ClassOrInterfaceType scope, final String name, final List<Type> typeArgs) {
```

Comment:
Perhaps instead of using a boolean parameter we could either:
* provide an enum with three values (TypeParamsNotSpecified, TypeParamsDerived, TypeParamsNotSpecified)
* make this constructor private and expose two static factory methods